### ISSUE TITLE
Check Bellatrix gossip verification

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#2984]
## Description

The Bellatrix specification [modifies gossip propagation conditions](https://github.com/ethereum/consensus-specs/blob/dev/specs/bellatrix/p2p-interface.md#beacon_block) for beacon blocks. We need to check that Lighthouse adheres to these conditions and add regression tests for all conditions.

This should be a simple task since I doubt there's much variance and there are [already tests](https://github.com/sigp/lighthouse/blob/stable/beacon_node/beacon_chain/tests/block_verification.rs) that check other gossip propagation conditions.

**Comment by paulhauner** [comment]
@pawanjay176 indicated he's keen to take this :relaxed:

**Comment by pawanjay176** [comment]
I think all the gossip verification conditions are already satisfied!

> If the execution is enabled for the block -- i.e. is_execution_enabled(state, block.body) then validate the following:
[REJECT] The block's execution payload timestamp is correct with respect to the slot -- i.e. execution_payload.timestamp == compute_timestamp_at_slot(state, block.slot).

Covered by:
https://github.com/sigp/lighthouse/blob/696de58141ecc34010c22256efac9c07849e5fb0/beacon_node/beacon_chain/src/execution_payload.rs#L178-L194


> If exection_payload verification of block's parent by an execution node is not complete:
[REJECT] The block's parent (defined by block.parent_root) passes all validation (excluding execution node verification of the block.body.execution_payload).
otherwise:
[IGNORE] The block's parent (defined by block.parent_root) passes all validation (including execution node verification of the block.body.execution_payload).

Here, the `parent_block` wouldn't be added to fork choice if it did not pass validation which covers both the cases.
https://github.com/sigp/lighthouse/blob/696de58141ecc34010c22256efac9c07849e5fb0/beacon_node/beacon_chain/src/execution_payload.rs#L163-L176



> The following gossip validation from prior specifications MUST NOT be applied if the execution is enabled for the block -- i.e. is_execution_enabled(state, block.body):
[REJECT] The block's parent (defined by block.parent_root) passes validation.

This also seems to be satisfied as we add the parent block to fork choice with `ExecutionStatus::Unknown` if the execution status is unknown.

TL;DR: I think this issue can be closed as we already pass all gossip verification conditions.

**Comment by paulhauner** [comment]
> TL;DR: I think this issue can be closed as we already pass all gossip verification conditions.

Great, thanks! I concur.
